### PR TITLE
Add actual footer tests

### PR DESCRIPTION
**Pull request by fabiosantoscode**
Footer needed some tests. Here they are.

### CODE REVIEW

File: test/index.es6 [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import Footer from '../index.es6';
+import List from '@economist/component-list';
+import Icon from '@economist/component-icon';
 import React from 'react/addons';
+import sinon from 'sinon';
```
Comment (umbobabo) [inline test/index.es6]: Sounds like a Team choice have another library regarding test, I'm not saying is wrong using Sinon but probably we should decide together if we need it or not.

File: test/index.es6 [hunk 2]
```diff
@@ -10,4 +13,67 @@ describe(`A Footer`, () => {
...
+      sinon.spy(footer, 'renderSocialListContent');
+
+      footer.render();
+
+      footer.renderListContent.called.should.equal(true);
+      footer.renderSocialListContent.calledOnce.should.equal(true);
+
+      footer.renderListContent.calledWith(basicData.customer).should.equal(true);
+      footer.renderListContent.calledWith(basicData.economist).should.equal(true);
+      footer.renderListContent.calledWith(basicData.business).should.equal(true);
```
Comment (umbobabo) [inline test/index.es6]: Why call 3 times the same function with the same value passed => [] ...?

Comment (umbobabo) [inline test/index.es6]: index.es6 doesn't seem to be linted.
Pls review and clarify my doubts.
Thank you.

Comment (fabiosantoscode) [inline test/index.es6]: 1. Different empty arrays were passed. They're !== from each other.
2. Yeah, gotta lint that.

File: test/index.es6 (hunk 1 above)
Comment (keithamus) [inline test/index.es6]: Precedent is that we've been using chai-spies (it's [already in your devdeps](https://github.com/fabiosantoscode/component-footer/blob/footer-tests/package.json#L142)). They integrate better with chai and are much simpler to grok. If there is a specific reason why you want to use Sinon, we should probably discuss this as a group, with a view to removing chai-spies from the devpack.

File: test/index.es6 [hunk 3]
```diff
@@ -10,4 +13,67 @@ describe(`A Footer`, () => {
...
+      footer.renderSocialListContent.calledWith(basicData.social).should.equal(true);
+    });
+    ['renderListContent', 'renderSocialListContent'].forEach((func) => {
```
Comment (keithamus) [inline test/index.es6]: Please don't add logic when defining tests. I know it seems not very dry, but this kind of thing makes tests far less readable. Also if the tests for these two functions are identical then maybe the functionality is - in which case why not just consolidate them?

File: test/index.es6 [hunk 4]
```diff
@@ -10,4 +13,67 @@ describe(`A Footer`, () => {
...
+          links.length.should.equal(1);
+          links[0].type.should.equal('a');
+          links[0].props.href.should.equal('http://example.com/6');
+          if (func === 'renderListContent') {
+            links[0].props.children.should.equal('6');
+          } else {
+            links[0].props.children.type.should.equal(Icon);
+            links[0].props.children.props.icon.should.equal('six');
+          }
+        });
```
Comment (keithamus) [inline test/index.es6]: Here the following could be done:

```js
links[0].should.deep.equal(
  <a href="http://example.com/6">
    <Icon icon="six" color="#B6B6B6"/>
  </a>
);
```

The benefit being that the above code is loads more readable, while still testing all the same stuff (and a tiny bit more).

File: test/index.es6 [hunk 5]
```diff
@@ -10,4 +13,67 @@ describe(`A Footer`, () => {
...
+            href: 'http://example.com/6',
+            title: '6',
+            meta: 'six',
+          }]);
+          links.length.should.equal(1);
+          links[0].type.should.equal('a');
+          links[0].props.href.should.equal('http://example.com/6');
+          if (func === 'renderListContent') {
+            links[0].props.children.should.equal('6');
+          } else {
```
Comment (keithamus) [inline test/index.es6]: See above comment. Adding this logic makes the tests very difficult to read & reason about.

**Comment by fabiosantoscode** [comment]
@keithamus and @umbobabo I've attempted to address all of your concerns. Using chai-spies now, no logic in tests, consolidated renderSocialListContents and renderListContents together.

File: test/index.es6 [hunk 6]
```diff
@@ -10,4 +12,91 @@ describe(`A Footer`, () => {
...
+        href: 'http://example.com/6',
+        title: '6',
+        meta: 'six',
+      }];
+      it('Returns an array of <a> tags', () => {
+        const links = footer.renderListContent(exampleLinks);
+        links.length.should.equal(1);
+        links[0].type.should.equal('a');
+        links[0].props.href.should.equal('http://example.com/6');
+        links[0].props.children.should.equal('6');
```
Comment (keithamus) [inline test/index.es6]: Still think this would be more readable if you just put some JSX here instead.

File: index.es6 [hunk 7]
```diff
@@ -15,37 +15,39 @@ export default class Footer extends React.Component {
     }
     return {};
   }
-  renderListContent(array) {
+  renderListContent(array, options = {}) {
+    const { useIcons = false, iconColor = '#B6B6B6' } = options;
```
Comment (keithamus) [inline index.es6]: Why not have:

```js
renderListContent(array, { useIcons = false, iconColor = '#B6B6B6' } = {}) {
}
```

Comment (fabiosantoscode) [inline index.es6]: I tried that, but babel compiled it incorrectly. Besides, if only one argument was passed, wouldn't iconColor be undefined?

**Pull request merged**